### PR TITLE
Don't skip indexing for old articles

### LINKED ISSUE #19039: Old articles doesn't indexing
**Describe the bug**

All old atricles removed from indexing
Example: https://ux.pub/editorial/kak-sozdat-bolshoie-razmytoie-izobrazhieniie-v-photoshop-illustrator-ili-sketch-4mod

We have moved all of the old content to the forem and we have articles with old post dates. Google start removing them from indexing after forem was updated

**To Reproduce**

1. Change "published at" of any article to a date prior to 2016.
2. Noindex added to article

image

**Expected behavior**

All articles, even old ones, are indexed

**Additional context**

Maybe caused by this commit
https://github.com/forem/forem/commit/c57acc25ea9814c1ba29e986673e26bf463eccaf#diff-24503fd25ed68e6ebceee4951bc4f9b255b197278d4aa9d86ef9d5afe3f26bea

All articles published before 14 July 2017 doesn't indexing? 
https://github.com/forem/forem/blob/main/app/models/article.rb
```
def skip_indexing?
    # should the article be skipped indexed by crawlers?
    # true if unpublished, or spammy,
    # or low score, not featured, and from a user with no comments
    !published ||
      (score < Settings::UserExperience.index_minimum_score &&
       user.comments_count < 1 &&
       !featured) ||
      published_at.to_i < 1_500_000_000 ||
      score < -1
  end
```

Thank you for your help!

### PR DESCRIPTION
**Pull request by dinozavrix**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Remove `published_at.to_i < 1_500_000_000 ||` from articles `skip_indexing` for indexing old articles 
Because of this condition old articles skip indexing
example: view-source:https://dev.to/bhaskar_vk/how-to-use-twitters-search-rest-api-most-effectively

## Related Tickets & Documents

- Related Issue #19039 
- Closes #19039 

## QA Instructions, Screenshots, Recordings

this condition skips indexing for all articles published before July 14, 2017
to check, you can open any articles before that date and check that these tags are not there
```
<meta name="robots" content="noindex">
<meta name="robots" content="nofollow">
```

example: view-source:https://dev.to/bhaskar_vk/how-to-use-twitters-search-rest-api-most-effectively

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [x] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [x] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -589,7 +589,6 @@ def skip_indexing?
       (score < Settings::UserExperience.index_minimum_score &&
        user.comments_count < 1 &&
        !featured) ||
-      published_at.to_i < 1_500_000_000 ||
```
Comment (jaw6) [inline app/models/article.rb:592]: :wave: Thanks for opening this PR and the original issue here! 

I wasn't immediately clear where this condition came from, so I spent a few minutes digging. It looks like it originated in [this PR](https://github.com/forem/forem/pull/14801/files#diff-24503fd25ed68e6ebceee4951bc4f9b255b197278d4aa9d86ef9d5afe3f26beaR501) from 2021, where it seems to be pretty deliberately de-indexing older articles. (It changes from `featured_number` to the current `published_at` in [this PR](https://github.com/forem/forem/pull/18510/files#diff-e45f0ec3372edbef08d5a1a7563b9e72dbd017794c286d90ca8c2fc2f434a982L117-R116), which was removing `featured_number` from articles anyway.)

I have the sense that some Forems really _do_ want to de-index older content, but some Forems do _not_. I wonder if a way to accommodate both use-cases might be to add a configurable setting?

**Review by jaw6 — CHANGES_REQUESTED** [review]

**Comment by rt4914** [comment]
Hi @dinozavrix 
Thanks for this PR. Can you reply to above [open comments](https://github.com/forem/forem/pull/19231#discussion_r1141920192) so that we can move ahead in this PR. If you have any questions, feel free to reach-out. 😃

**Comment by jaw6** [comment]
Hey @dinozavrix — The way I would approach this would be something like:

* Start by defining the new setting in [user experience settings](https://github.com/forem/forem/blob/main/app/lib/constants/settings/user_experience.rb) — use the [current value](https://github.com/forem/forem/blob/main/app/models/article.rb#L597) `1_500_000_000` as the **default** for the new setting
* Add a field for the new setting to [the user experience form](https://github.com/forem/forem/blob/main/app/views/admin/settings/forms/_user_experience.html.erb)
* Use the new setting in [the `skip_indexing?` check](https://github.com/forem/forem/blob/main/app/models/article.rb#L594-L597)

There's a few pieces to the puzzle, but overall, it should be pretty straight-forward!

**Review by jaw6 — CHANGES_REQUESTED** [review]
See [comment](https://github.com/forem/forem/pull/19231#issuecomment-1523290165)

**Comment by dinozavrix** [comment]
Hi @jaw6, thanks for your help! Here is the updated commit
https://github.com/forem/forem/pull/19231/commits/5a3b9c337432e712f664f911972136377c44d6f3

**Review by jaw6 — APPROVED** [review]
This looks great! :+1:

**Comment by rt4914** [comment]
Hi @Ridhwana and @maestromac Please have a look at this PR whenever possible.

**Comment by dinozavrix** [comment]
@Ridhwana thanks for you help!

**Pull request merged**